[Web] Make language on mfa verify step dialog more clear

rudream opened the pull request:
## Purpose

This PR resolves https://github.com/gravitational/teleport/issues/20636

Makes the instructions on the verify dialog more clear to specify that you must first verify with an already registered device.

## Demo

<img width="520" alt="image" src="https://user-images.githubusercontent.com/56373201/215137276-0dad652c-30b4-4fd7-9e05-2257256919fd.png">

rudream committed:
update text on dialog

zmb3 commented:
Let's run the language by @roraback.

Kenny - when registering a _new_ MFA device, we first need to prompt the user to tap their _existing_ MFA device. This wasn't clear to users and they would tap the new MFA device first and get an error since it isn't yet registered.

ravicious commented on web/packages/teleport/src/components/ReAuthenticate/ReAuthenticate.tsx:
```diff
@@ -78,7 +78,8 @@ export function ReAuthenticate({
           <DialogHeader style={{ flexDirection: 'column' }}>
```
I noticed that pressing Enter after typing the OTP doesn't submit the form. AFAIR, a quick fix for that would be to wrap the body of this component in `<form>` and set the type of the "Continue" button to `submit`.

ravicious reviewed (approved):
This is a step in the right direction but I feel like to completely remove the ambiguity, something more is needed. I ran into this myself probably a dozen of times and I don't think a slight change in copy would prevent me from making this mistake again. :D

Honestly, long term we should probably improve the error message if possible, into something like "lol, this is the device you're trying to _remove_, please use another one to make sure you don't lose access after removing xyz".

roraback commented:
1. If we need this to be a simple, generic message that can work in any context, including outside the "Account Settings" screen, I think I would tweak the language slightly to:

> You must verify your identity with one of your existing two-factor devices before performing this action.

2. A bit better, if it's possible/easy to give context of the action they're taking:
> You must verify your identity with one of your **existing** two-factor devices before registering a new device.

3. Even better, if it's possible to also give context of what two-factor devices they already registered:
If user has one device:
> You must verify your identity with your registered [insert device type] “[insert name]” before registering a new device.

Else if user has multiple devices:
> You must verify your identity with one of the following existing two-factor devices before registering a new device:
[for each registered key]
>* [device type], "[name]"

rudream committed:
tweak language and overwrite error message

rudream commented:
Changes:

- Added an optional `dialogText` prop that allows parents to provide custom text to show in the dialog depending on the context.
- Increased the width of the dialog slightly to prevent the word "two-factor" from being broken up into two lines making it harder to read.
- Fixed the form not submitting when pressing `Enter` for TOTP

> Honestly, long term we should probably improve the error message if possible, into something like "lol, this is the device you're trying to remove, please use another one to make sure you don't lose access after removing xyz".

@ravicious The error is a frontend `DOMException` error that comes from webauthn and is not an error from a backend response that we can tweak, it also only shows up on Firefox. The only way I could think to make it more descriptive is to manually overwrite that error message, please check my latest commit and let me know what you think of this approach.

ravicious commented on web/packages/teleport/src/components/ReAuthenticate/ReAuthenticate.tsx:
```diff
@@ -69,68 +70,71 @@ export function ReAuthenticate({
+            <DialogHeader style={{ flexDirection: 'column' }}>
+              <DialogTitle>Verify your identity</DialogTitle>
+              <Text textAlign="center" color="text.secondary">
+                {dialogText}
```
Super nit: I'd consider hardcoding the part that doesn't change ("You must verify your identity with one of your existing two-factor devices before") and letting the callsites change only the last part.

ravicious commented on web/packages/teleport/src/components/ReAuthenticate/useReAuthenticate.ts:
```diff
@@ -51,13 +68,18 @@ export default function useReAuthenticate({ onAuthenticated, onClose }: Props) {
     submitWithWebauthn,
     auth2faType: cfg.getAuth2faType(),
     preferredMfaType: cfg.getPreferredMfaType(),
+    dialogText,
```
I'd pass this directly from `Container`'s props to `ReAuthenticate` without it going through the hook, but I don't want to barge in with new practices in an arbitrary PR.

rudream replied:
For the time being I think the current approach keeps our type cleaner, we will be shifting away from using `Container` with hooks in the future anyways.

ravicious reviewed (approved):
Heh, I booted up Chrome to see how it handles the problem compared to Firefox and I ran into the issue this PR is trying to solve _again_ while trying to set up a new hardware key and just mindlessly clicking around. :D

ravicious commented:
BTW, your change did fix keyboard navigation in ReAuthenticate though I notice that usually the next dialog that shows up after it (when removing or adding a device) doesn't handle Enter properly. It'd be good clean that up at some point.

rudream committed:
address comments

rudream committed:
fix 'enter' to submit on newdevice form

rudream committed:
Merge branch 'master' into yassine/web/mfa-verify-dialog

rudream committed:
Merge branch 'master' into yassine/web/mfa-verify-dialog

rudream committed:
Merge branch 'master' into yassine/web/mfa-verify-dialog

rudream committed:
Merge branch 'master' into yassine/web/mfa-verify-dialog
